sale_order_client_order_ref_mandatory/views/sale_order_view.xml

```diff
@@ -0,0 +1,21 @@
+<?xml version="1.0" encoding="utf-8"?>
+<odoo>
```

Comment:
Could you add `client_order_ref` in search view, and in tree view as `optional=show`, after `partner_id` field?